src/zoo.js

```diff
@@ -11,57 +11,137 @@ eslint no-unused-vars: [
+  const bichos = {};
+  const filterAnimal = element => (bichos[element.name] = element.residents.length);
+  data.animals.filter(filterAnimal);
+  return (species) ? bichos[species] : bichos;
+};
+
+const entryCalculator = function (entrants) {
+  return !entrants || Object.keys(entrants).length === 0 ?
+    0
+    :
+    (data.prices.Adult * entrants.Adult) +
+    (data.prices.Child * entrants.Child) +
+    (data.prices.Senior * entrants.Senior);
+};
+
+const createNewObj = (animal, options) => {
+  const newObj = {};
+  if (options.sex === 'female' || options.sex === 'male') {
+    const array = [];
+    animal.residents.forEach((element) => {
+      if (element.sex === options.sex) {
+        array.push(element.name);
+      }
+    });
+    newObj[animal.name] = array;
+  } else {
+    newObj[animal.name] = animal.residents.map(element => element.name);
+  }
+  if (options.sorted) {
+    newObj[animal.name].sort();
+  }
+  return newObj;
+};
 
-function isManager(id) {
-  // seu código aqui
-}
+const animalMap = function (options) {
+  const obj = { NE: [], NW: [], SE: [], SW: [] };
 
-function addEmployee(id, firstName, lastName, managers, responsibleFor) {
-  // seu código aqui
-}
+  if (!options || !options.includeNames) {
+    data.animals.map(animal => obj[animal.location].push(animal.name));
+  } else {
+    data.animals.map(animal => obj[animal.location].push(createNewObj(animal, options)));
+  }
 
-function animalCount(species) {
-  // seu código aqui
-}
+  return obj;
+};
 
-function entryCalculator(entrants) {
-  // seu código aqui
-}
+const schedule = function (dayName) {
+  const keys = Object.keys(data.hours);
+  const obj = {};
+  keys.forEach(function (cur) {
+    if (data.hours[cur].open === 0) {
+      obj[cur] = 'CLOSED';
+    } else {
+      obj[cur] = `Open from ${data.hours[cur].open}am until ${data.hours[cur].close - 12}pm`;
+    }
+  });
+  const objFiltered = {};
+  objFiltered[dayName] = obj[dayName];
 
-function animalMap(options) {
-  // seu código aqui
-}
+  return (dayName) ? objFiltered : obj;
+};
 
-function schedule(dayName) {
-  // seu código aqui
-}
+const oldestFromFirstSpecies = (id) => {
+  const idAnimal = data.employees.filter(employee => employee.id === id)[0].responsibleFor[0];
+  const Animal = data.animals.filter(animal => animal.id === idAnimal);
+  const olderAnimal = Object.values(Animal[0].residents.sort((a, b) => b.age - a.age)[0]);
+  return olderAnimal;
+};
 
-function oldestFromFirstSpecies(id) {
-  // seu código aqui
-}
+const increasePrices = (percentage) => {
+  const adult = data.prices.Adult;
+  const senior = data.prices.Senior;
+  const child = data.prices.Child;
 
-function increasePrices(percentage) {
-  // seu código aqui
-}
+  data.prices.Adult = Math.round(adult * ((percentage / 100) + 1) * 100) / 100;
+  data.prices.Senior = Math.round(senior * ((percentage / 100) + 1) * 100) / 100;
+  data.prices.Child = Math.round(child * ((percentage / 100) + 1) * 100) / 100;
```

Comment:
Essa parte aqui pra dar uma diminuida nas linhas, usando _array_ e _forEach_, ai não precisaria declarar essas _consts_ ali em cima. Algo tipo:
```suggestion
Object.keys(data.prices).forEach((price) => {
    (data.prices[price] = Math.round(data.prices[price] *
    ((percentage / 100) + 1) * 100) / 100);
```